### PR TITLE
Add soviet-06a

### PR DESCRIPTION
**Pull request by ghost**
This adds the mission soviet-06a.

**Changes compared to the original:**
- Removed soviet tech center as it doesn't provide the player with unlocks (flamethrower)

**Known issues:**
- Name is too long for the mission browser (Shorten the name or rework the browser to support it?)

**Suggestions for difficulty settings are more than welcome.**

### CODE REVIEW

**Comment by MustaphaTR** [comment]
I played the map and couldn't see any problem. I think we should make mission browser bigger, because currently RA one is smallest of all, so resizing shouldn't be a problem on small resolutions.

File: mods/ra/maps/soviet-06a/soviet06a-AI.lua [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	end)
+end
+
+SendUnits = function(units, waypoints)
+	Utils.Do(units, function(unit)
+		if not unit.IsDead then
+			Utils.Do(waypoints, function(waypoint)
+				unit.AttackMove(waypoint.Location)
+			end)
+			unit.Hunt()
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a-AI.lua]: I'd suggest using `IdleHunt` here.

File: mods/ra/maps/soviet-06a/soviet06a.lua [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+Trigger.OnRemovedFromWorld(Mcv, function()
+	if not mcvDeployed then
+		mcvDeployed = true
+		BuildBase()
+		Trigger.AfterDelay(DateTime.Minutes(1), ProduceInfantry)
+		Trigger.AfterDelay(DateTime.Minutes(2), ProduceArmor)
+		Trigger.AfterDelay(DateTime.Minutes(2), function()
+			Utils.Do(BaseAttackers, function(actor)
+				if not actor.IsDead then
+					Trigger.OnIdle(actor, actor.Hunt)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a.lua]: I think you can use `IdleHunt` here too.

File: mods/ra/maps/soviet-06a/soviet06a.lua [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+end)
+
+WorldLoaded = function()
+	player = Player.GetPlayer("USSR")
+	enemy = Player.GetPlayer("Greece")
+	Camera.Position = CameraStart.CenterPosition
+	if not Mcv.IsDead then
+		Mcv.Move(McvWaypoint.Location)
+	end
+	if not Harvester.IsDead then
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a.lua]: Why should units be dead at the start of the mission?

File: mods/ra/maps/soviet-06a/soviet06a.lua [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		Harvester.FindResources()
+	end
+	Utils.Do(IntroAttackers, function(actor)
+		if not actor.IsDead then
+			Trigger.OnIdle(actor, actor.Hunt)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a.lua]: `IdleHunt`?

File: mods/ra/maps/soviet-06a/soviet06a.lua [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+		if actor.Owner == enemy and actor.HasProperty("StartBuildingRepairs") then
+			Trigger.OnDamaged(actor, function(building)
+				if building.Owner == enemy and building.Health < 3/4 * building.MaxHealth then
+					building.StartBuildingRepairs()
+				end
+			end)
+		end
+	end)
+	Reinforcements.ReinforceWithTransport(player, "apc", SovietReinforcements1, SovietReinforcements1Waypoints, nil, function(apc, team)
+		table.insert(team, apc)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a.lua]: What does this do?
(Same for below.)

File: mods/ra/maps/soviet-06a/soviet06a.lua (hunk 3 above)
Comment (Biofreak1987) [inline mods/ra/maps/soviet-06a/soviet06a.lua]: Cause you told me so to check units at the start of the mission in soviet03 :p

File: mods/ra/maps/soviet-06a/soviet06a.lua (hunk 5 above)
Comment (Biofreak1987) [inline mods/ra/maps/soviet-06a/soviet06a.lua]: I've reused it from gdi04a, maybe you could tell me what it's used for in there?

**Comment by abcdefg30** [comment]
What about a secondary objective like `Eliminate the allied presence in this area.`?

To difficulties: You could add allied reinforcements through chinooks / water transports on harder difficulties. And/Or also increase the attack group count and decrease the delay between the build functions.

Anyway, it already looks good as is. :+1: / :white_check_mark:

**Comment by Biofreak1987** [comment]
Updated, just discovered by replaying the mission that you could sell the trucks at the mcv.

File: mods/ra/maps/soviet-06a/soviet06a.lua [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+		Trigger.AfterDelay(DateTime.Minutes(2), ProduceArmor)
+		Trigger.AfterDelay(DateTime.Minutes(2), function()
+			Utils.Do(BaseAttackers, function(actor)
+				IdleHunt(actor)
+			end)
+		end)
+	end
+end)
+
+IdleHunt = function(actor)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a.lua]: This shouldn't be necessary as you can access the `IdleHunt` function from `soviet06a-AI.lua`.

File: mods/ra/maps/soviet-06a/soviet06a-AI.lua (hunk 1 above)
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a-AI.lua]: You missed this one, as it seems.

File: mods/ra/maps/soviet-06a/soviet06a.lua [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+	Utils.Do(Map.NamedActors, function(actor)
+		if actor.Owner == enemy and actor.HasProperty("StartBuildingRepairs") then
+			Trigger.OnDamaged(actor, function(building)
+				if building.Owner == enemy and building.Health < 3/4 * building.MaxHealth then
+					building.StartBuildingRepairs()
+				end
+			end)
+		end
+	end)
+	Reinforcements.ReinforceWithTransport(player, "apc", SovietReinforcements1, SovietReinforcements1Waypoints, nil, function(apc, team)
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a.lua]: I don't think you don't need (here and below) the last two parameters.

**Comment by abcdefg30** [comment]
If you have fixed/answered to the last three remaining nits and squashed the commits, this is good to go.

**Comment by Biofreak1987** [comment]
Done

File: mods/ra/maps/soviet-06a/soviet06a-AI.lua (hunk 1 above)
Comment (abcdefg30) [inline mods/ra/maps/soviet-06a/soviet06a-AI.lua]: Sorry, you still missed this. Or is it intentional?

**Comment by abcdefg30** [comment]
Just waiting for https://github.com/OpenRA/OpenRA/pull/9656#discussion_r42319978.

File: mods/ra/maps/soviet-06a/soviet06a-AI.lua (hunk 1 above)
Comment (Biofreak1987) [inline mods/ra/maps/soviet-06a/soviet06a-AI.lua]: I missed that one, I'll update both 6a and 6b shortly.

**Comment by Biofreak1987** [comment]
K, now that one should be IdleHunt as well.

**Comment by abcdefg30** [comment]
Good work with both missions!

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1b3a9ed13cf2c8c61857cb12e6201ff8015fea39...937535edd1cc037f8e07d85e6982bd07997b7d17)

**Pull request merged**